opencog/atomspace/AtomTable.cc

```diff
@@ -284,6 +284,9 @@ AtomPtr AtomTable::factory(Type atom_type, AtomPtr atom)
     } else if (BIND_LINK == atom_type) {
         if (NULL == BindLinkCast(atom))
             return createBindLink(*LinkCast(atom));
+    } else if (PATTERN_LINK == atom_type) {
+        if (NULL == PatternLinkCast(atom))
+            return createPatternLink(*LinkCast(atom));
```

Comment:
This probably should not be done, although the rest of the patch seems OK. PatternLink is meant to be a base-class, and not actually instantiated.